### PR TITLE
[Twitch] Add support for player.twitch.tv urls

### PR DESCRIPTION
**Pull request by rob2001**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Addresses issue #11535 by modifying the VOD (video on demand) regex pattern in the Twitch Information Extractor to be compatible with urls of the form "player.twitch.tv/?video=v12345...."

### CODE REVIEW

File: youtube_dl/extractor/twitch.py [hunk 1]
```diff
@@ -230,6 +231,26 @@ class TwitchVodIE(TwitchItemBaseIE):
             'skip_download': True,
         },
     }, {
+        # player.twitch.tv URL
+        'url': 'http://player.twitch.tv/?video=v6528877&t=5m10s',
```
Comment (dstftw) [inline youtube_dl/extractor/twitch.py]: Test should be `only_matching`.

File: youtube_dl/extractor/twitch.py [hunk 2]
```diff
@@ -206,7 +206,8 @@ class TwitchChapterIE(TwitchItemBaseIE):
 
 class TwitchVodIE(TwitchItemBaseIE):
     IE_NAME = 'twitch:vod'
-    _VALID_URL = r'%s/[^/]+/v/(?P<id>\d+)' % TwitchBaseIE._VALID_URL_BASE
+    _VALID_URL_BASE = r'https?://(?:www\.|player\.)?twitch\.tv'
```
Comment (dstftw) [inline youtube_dl/extractor/twitch.py]: What's the point extracting this?

File: youtube_dl/extractor/twitch.py [hunk 3]
```diff
@@ -206,7 +206,8 @@ class TwitchChapterIE(TwitchItemBaseIE):
...
+    _VALID_URL = r'%s/(?:[^/]+/v/|\?video=v)(?P<id>\d+)' % _VALID_URL_BASE
```
Comment (dstftw) [inline youtube_dl/extractor/twitch.py]: Does not match http://player.twitch.tv/?t=5m10s&video=v6528877.
Should not match https://www.twitch.tv/?video=v6528877, http://player.twitch.tv/riotgames/v/6528877?t=5m10s.

File: youtube_dl/extractor/twitch.py (hunk 2 above)
Comment (rob2001) [inline youtube_dl/extractor/twitch.py]: I wasn't sure if the _VALID_URL_BASE from the BaseIE class was ever used elsewhere, so I updated it in TwichVodIE just to keep things consistent. I can remove it if it's not necessary.

**Comment by rob2001** [comment]
@dstftw Thanks for fixing the errors, I was busy this past week and wasn't able to get to them.

**Pull request closed without merging**